WordPress/gutenberg
# Avoid copying global style presets via the styles compatibility hook

## What?
Closes #52155.

This is just a PoC PR. It updates the `useCompatibilityStyles` hook to avoid copying styles directly rendered in the editors.

## Why?
* When the non-iframed editor is rendered, global style presets are injected into a `body`.
* The `useCompatibilityStyles` parses and memoized document stylesheet on initial load. So these styles are added to the compat styles array.
* The presets are copied as part of the compatibility layer when the iframed editor is rendered via pattern preview or switching device type. This results in "broken" previews and unnecessary warnings in the console.

## How?
~~I added a custom data attribute to the `style` tags generated by the `EditorStyles` and `LayoutStyle` components. This "marked" is used to skip collecting preset styles in the `compatStyles` list.~~

Skip the styles without IDs. Styles enqueued via the WP dependency system will always have IDs.

## Testing Instructions
1. Open a Post or Page.
2. Enable "Custom fields" from Preferences.
3. Switch to device preview 'Tablet' and confirm no warnings in the console.
4. Open the pattern inserter and switch to the "Featured" category.
5. Confirm pattern previews are rendered without extra spacing.

## Screenshots or screencast 
![CleanShot 2023-07-14 at 17 15 56](https://github.com/WordPress/gutenberg/assets/240569/e8cae52f-25ed-4cee-be60-8e201cc25cb5)
